lib/mpl_toolkits/axisartist/axes_grid.py

```diff
@@ -1,19 +1,13 @@
-from __future__ import (absolute_import, division, print_function,
-                        unicode_literals)
-
 import mpl_toolkits.axes_grid1.axes_grid as axes_grid_orig
 from .axes_divider import LocatableAxes
 
+
 class CbarAxes(axes_grid_orig.CbarAxesBase, LocatableAxes):
-    def __init__(self, *kl, **kwargs):
-        orientation=kwargs.pop("orientation", None)
-        if orientation is None:
-            raise ValueError("orientation must be specified")
+    def __init__(self, *args, orientation, **kwargs):
```

Comment:
Not exactly the same: You can now explicitly pass orientation=None and it will not raise a ValueError.

Reply:
`orientation="foobar"` was probably also invalid to start with, so there must be a downstream check that will also prune None out.

Reply:
Maybe you're right. I don't follow the code path here. And I couldn't find out whose responsibility checking or `orientation` really is.

When doing
~~~
ImageGrid(plt.figure(), (0, 0, 1,1), (1, 2), cbar_mode='single', cbar_location='foobar')
~~~

it finally raises a `KeyError` on `ax.axis[self.orientation].toggle(all=b)` in `CbarAxesBase._config_axes()`. However that seems more by chance than an intended check.

Before that we have code executed like

~~~
        if self.orientation in ["top", "bottom"]:
            orientation = "horizontal"
        else:
            orientation = "vertical"
~~~

which is locally coercing all invalid values to 'vertical'.

The whole thing seems a little brittle to me. But this is beyond this PR. By me, it's ok to not reject orientation=None in this case. Some code downstream will complain.